test/e2e/node_drain.go

```diff
@@ -233,6 +233,31 @@ func NodeDrainTimeoutSpec(ctx context.Context, inputGetter func() NodeDrainTimeo
 				WaitForDeploymentAvailableInterval: input.E2EConfig.GetIntervals(specName, "wait-deployment-available"),
 			})
 		}
+		By("Deploy an unevictable Deployment with `wait-completed` label.")
+		framework.DeployPodAndWait(ctx, framework.DeployPodAndWaitInput{
+			WorkloadClusterProxy: workloadClusterProxy,
+			ControlPlane:         controlplane,
+			DeploymentName:       cpDeploymentName("wait-completed"),
+			Namespace:            "unevictable-workload",
+			NodeSelector:         map[string]string{nodeOwnerLabelKey: "KubeadmControlPlane-" + controlplane.Name},
+			ModifyDeployment: func(deployment *appsv1.Deployment) {
+				deployment.Spec.Template.Labels["cluster.x-k8s.io/drain"] = string(clusterv1.MachineDrainRuleDrainBehaviorWaitCompleted)
```

Comment:
```suggestion
				deployment.Spec.Template.Labels["cluster.x-k8s.io/drain"] = "wait-completed"
```
nit: I know both works, but maybe let's test the one that we document in the API (same in l.256)